backend/src/LibExecution/ProgramTypes.fs

```diff
@@ -192,6 +192,8 @@ type MatchPattern =
 
   | MPVariable of id * string
 
+  | MPOr of id * List<MatchPattern>
```

Comment:
```suggestion
  // CLEANUP consider making this `List` an `NEList`
  | MPOr of id * List<MatchPattern>
```

I recognize that you experimented with making `MatchCase` allow for an `NEList<MP>`, which is similar to this, but what do you think about this^?

My concern with the current approach is that it allows for an `MPOr []` -- so there can be a match case without _any_ real patterns on the LHS. That feels like it should be illegal. We could either tackle that now, or later. Would the NEList being _here_ vs 'higher up' (i.e. `MatchCase`) make any difference?

Reply:
I think it will fail during the parsing phase before getting to this point, but it's probably worth updating it.
I don’t think there is much difference, but I still wonder if it would be cleaner to make the changes in MatchCase